File: tests/import.hs
module Import
where
data TestsImportImportedLib = TestsImportImportedLib
    { importedLibArrayFromImportedLib' :: [String]
    , importedLibParameterFromImportedLib' :: Integer
    }
data TestsImport = TestsImport
    { importedLib' :: TestsImportImportedLib
    , importedParameter' :: Integer
    , array' :: [Integer]
    }
testsImport :: TestsImport
testsImport = TestsImport
    { importedLib' = TestsImportImportedLib
        { importedLibArrayFromImportedLib' = ["a", "b"]
        , importedLibParameterFromImportedLib' = 21
        }
    , importedParameter' = 42
    , array' = [1, 4, 9, 16, 25, 36, 49, 64, 81, 100]
    }

File: tests/import.c
#include "import.h"
const struct t_tests_import TESTS_IMPORT = {
    .imported_lib =
        {
            .array_from_imported_lib = {[0] = "a", [1] = "b"},
            .parameter_from_imported_lib = 21,
        },
    .imported_parameter = 42,
    .array = {[0] = 1, [1] = 4, [2] = 9, [3] = 16, [4] = 25, [5] = 36, [6] = 49, [7] = 64, [8] = 81, [9] = 100},
};
